File: codechef/FEB19B/HMAPPY.cpp
// Suraj Pawar- ssp4all

#include <bits/stdc++.h>
# define lld long long int
int main() {
	int t;
	lld n, a, b, k, i;
	scanf("%d",&t);
	while(t--){
		lld pa = 0;
		lld pb = 0;
		scanf("%lld %lld %lld %lld",&n,&a,&b,&k);
		for(i=1; i<(n+1); i++){
			if(i%a == 0 && i%b == 0){
			}
			else if(i%a == 0 && i%b != 0)
				pa++;
			else if(i%a != 0 && i%b == 0)
				pb++;
			if((pa+pb)>=k){
				printf("Win\n");
				break;
			}
		}
		if((pa+pb)<k)
			printf("Lose\n");
	}
}

File: codechef/FEB19B/HMAPPY.py
# cook your dish here
# Suraj Pawar- ssp4all
t = int(input())
for _ in range(t):
	n, a, b, k = map(int, input().split())
	pa, pb = 0, 0
	for i in range(1,n+1):
		if i%a == 0 and i%b == 0:
			continue
		elif i%a == 0 and i%b != 0:
			pa += 1
		elif i%b == 0 and i%a != 0:
			pb += 1
	if (pa+pb) >= k:
		print('Win')
	else:
		print('Lose')